pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -257,6 +268,8 @@ public BrokerService(PulsarService pulsar) throws Exception {
                 .newSingleThreadScheduledExecutor(new DefaultThreadFactory("pulsar-msg-expiry-monitor"));
         this.compactionMonitor =
             Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("pulsar-compaction-monitor"));
+        this.messagePublishBufferMonitor =
```

Comment:
We should create this executor only when the feature is enabled.

Also I see we are creating more and more schedulers. Can we consider reusing some of the executors?

Reply:
Hmm, I think we need a different thread name. It's better for jstack analysis.